### PR TITLE
ZL41 Board sambles switch libstm32 -> libperiph

### PR DESCRIPTION
**Pull request by lucckb**
Switch ZL41 board for old libstm32 to -> libperiph

### CODE REVIEW

File: zl41arm/blink/blinkf4.cpp [hunk 1]
```diff
@@ -1,140 +1,57 @@
...
+		nullptr,
+		[]() {
+			m_ulock_sem.wait(ISIX_TIME_INFINITE);
+		},
+		[]() {
+			m_ulock_sem.signal();
+		},
+		periph::drivers::uart_early::open,
+		"serial0", 115200
+	);
```
Comment (tomeksowi) [inline zl41arm/blink/blinkf4.cpp]: mixed indent, tabs, most of this file has spaces.

File: zl41arm/board/include/isix_config.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
+ * =====================================================================================
+ *
+ *       Filename:  config.h
```
Comment (tomeksowi) [inline zl41arm/board/include/isix_config.h]: Doesn't match the filename. Maybe remove these headers altogether? They don't bring any useful info anyway.

File: zl41arm/board/platform_startup.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+bool uc_periph_setup()
+{
+
+	constexpr auto retries=100000;
+
+	isix_set_irq_vectors_base( &_exceptions_vectors );
+
+    //! Deinitialize RCC
+    LL_RCC_DeInit();
```
Comment (tomeksowi) [inline zl41arm/board/platform_startup.cpp]:
```suggestion
	//! Deinitialize RCC
	LL_RCC_DeInit();
```

File: zl41arm/board/platform_startup.cpp [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	LL_RCC_PLL_Enable();
+	for( auto r=0; r<retries; ++r ) {
+		if( LL_RCC_PLL_IsReady() ) {
+			break;
+		}
+	}
+	if( !LL_RCC_PLL_IsReady() ) {
+		return false;
+	}
+    LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_PLL);
```
Comment (tomeksowi) [inline zl41arm/board/platform_startup.cpp]:
```suggestion
	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_PLL);
```

File: zl41arm/dacaudio/dac_audio.cpp [hunk 5]
```diff
@@ -81,51 +81,72 @@ dac_audio::dac_audio( size_t audio_buf_len, size_t num_buffers )
 dac_audio::~dac_audio()
 {
 	do_stop();
-	stm32::nvic_irq_enable( DAC_DMA_STREAM_IRQN, false );
-	stm32::rcc_apb1_periph_clock_cmd( RCC_APB1Periph_DAC|AUDIO_TIMER_RCC, false );
+isix::free_irq( DAC_DMA_STREAM_IRQN);
+    LL_APB1_GRP1_DisableClock(LL_APB1_GRP1_PERIPH_DAC1|AUDIO_TIMER_RCC);
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_audio.cpp]:
```suggestion
	isix::free_irq( DAC_DMA_STREAM_IRQN);
	LL_APB1_GRP1_DisableClock(LL_APB1_GRP1_PERIPH_DAC1|AUDIO_TIMER_RCC);
```

File: zl41arm/dacaudio/dac_audio.cpp [hunk 6]
```diff
@@ -81,51 +81,72 @@ dac_audio::dac_audio( size_t audio_buf_len, size_t num_buffers )
...
+    LL_DAC_InitTypeDef dac = {
+        .TriggerSource = DAC_TRIGER,
+        .WaveAutoGeneration = LL_DAC_WAVE_AUTO_GENERATION_NONE,
+        .WaveAutoGenerationConfig = 0,
+        .OutputBuffer = LL_DAC_OUTPUT_BUFFER_ENABLE
+    };
+	LL_DAC_Init( DAC1,DAC_CHANNEL ,&dac);
+    isix::set_irq_priority( DAC_DMA_STREAM_IRQN, {IRQ_PRIO, IRQ_SUB} );
+    isix::request_irq( DAC_DMA_STREAM_IRQN );
 	return err_success;
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_audio.cpp:117]:
```suggestion
    return err_success;
```

File: zl41arm/dacaudio/dac_audio.cpp [hunk 7]
```diff
@@ -81,51 +81,72 @@ dac_audio::dac_audio( size_t audio_buf_len, size_t num_buffers )
...
 
-//! Setup the hardware 
-int dac_audio::hardware_setup() 
+//! Setup the hardware
+int dac_audio::hardware_setup()
 {
-	using namespace stm32;
-	rcc_ahb1_periph_clock_cmd( RCC_AHB1Periph_GPIOA|DAC_DMA_RCC, true );
-	rcc_apb1_periph_clock_cmd( RCC_APB1Periph_DAC|AUDIO_TIMER_RCC, true );
+	LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_GPIOA|DAC_DMA_RCC);
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_audio.cpp]:
```suggestion
    LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_GPIOA|DAC_DMA_RCC);
```

File: zl41arm/dacaudio/dac_audio.cpp [hunk 8]
```diff
@@ -81,51 +81,72 @@ dac_audio::dac_audio( size_t audio_buf_len, size_t num_buffers )
...
-	stm32::tim_timebase_init( AUDIO_TIMER, 0, TIM_CounterMode_Up, val, 0, 0 );
-	stm32::tim_select_output_trigger( AUDIO_TIMER, TIM_TRGOSource_Update );
+    LL_TIM_InitTypeDef timcfg = {
+        .Prescaler = 0,
+        .CounterMode = LL_TIM_COUNTERMODE_UP,
+        .Autoreload = val,
+        .ClockDivision = LL_TIM_CLOCKDIVISION_DIV1,
+        .RepetitionCounter =  0
+    };
+LL_TIM_Init( AUDIO_TIMER, &timcfg );
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_audio.cpp]:
```suggestion
    LL_TIM_Init( AUDIO_TIMER, &timcfg );
```

File: zl41arm/dacaudio/dac_audio.cpp [hunk 9]
```diff
@@ -81,51 +81,72 @@ dac_audio::dac_audio( size_t audio_buf_len, size_t num_buffers )
...
+    LL_TIM_SetTriggerOutput( AUDIO_TIMER, LL_TIM_TRGO_UPDATE );
 	m_state = state::start_wait;
 	return m_error;
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_audio.cpp]:
```suggestion
    m_state = state::start_wait;
    return m_error;
```

File: zl41arm/dacaudio/dac_audio.cpp [hunk 10]
```diff
@@ -226,8 +261,8 @@ int dac_audio::commit_buffer( uint16_t* buf, ostick_t tout )
 		dbprintf("Buffer curently not playing audio %i", m_state );
 		m_error = err_not_playing;
 		return m_error;
-	}
-	if( m_error == ISIX_EOK ) { 
+    }
+	if( m_error == ISIX_EOK ) {
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_audio.cpp]:
```suggestion
    if( m_error == ISIX_EOK ) {
```

File: zl41arm/dacaudio/dac_test.cpp [hunk 11]
```diff
@@ -57,23 +57,37 @@ void dac_gen( void* )
...
-//! Main core function tester
-int main() {
 
-	dblog_init( stm32::usartsimple_putc, NULL, stm32::usartsimple_init,
-	    		USART1, 115200, false, CONFIG_PCLK1_HZ, CONFIG_PCLK2_HZ );
+//! Main core function tester
+int main()
+{
+    static isix::semaphore m_ulock_sem { 1, 1 };
+    isix::wait_ms( 500 );
```
Comment (tomeksowi) [inline zl41arm/dacaudio/dac_test.cpp]:
```suggestion
	static isix::semaphore m_ulock_sem { 1, 1 };
	isix::wait_ms( 500 );
```

**Review by tomeksowi — CHANGES_REQUESTED** [review]
Looks ok. Minor formatting

File: zl41arm/board/include/isix_config.h (hunk 2 above)
Comment (lucckb) [reply zl41arm/board/include/isix_config.h]: fixed

**Pull request merged**